server/src/main/java/org/opensearch/indices/tiering/TieringRequestValidator.java

```diff
@@ -0,0 +1,297 @@
+import org.opensearch.cluster.metadata.IndexMetadata;
+import org.opensearch.cluster.node.DiscoveryNode;
+import org.opensearch.cluster.routing.IndexRoutingTable;
+import org.opensearch.cluster.routing.ShardRouting;
+import org.opensearch.cluster.routing.allocation.DiskThresholdSettings;
+import org.opensearch.common.collect.Tuple;
+import org.opensearch.core.index.Index;
+import org.opensearch.index.IndexModule;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static org.opensearch.index.IndexModule.INDEX_TIERING_STATE;
+
+/**
+ * Validator class to validate the tiering requests of the index
+ * @opensearch.experimental
+ */
+public class TieringRequestValidator {
+
+    private static final Logger logger = LogManager.getLogger(TieringRequestValidator.class);
+
+    /**
+     * Validates the tiering request for indices going from hot to warm tier
+     *
+     * @param currentState current cluster state
+     * @param concreteIndices set of indices to be validated
+     * @param clusterInfo the current nodes usage info for the cluster
+     * @param diskThresholdSettings the disk threshold settings of the cluster
+     * @return result of the validation
+     */
+    public static TieringValidationResult validateHotToWarm(
+        final ClusterState currentState,
+        final Set<Index> concreteIndices,
+        final ClusterInfo clusterInfo,
+        final DiskThresholdSettings diskThresholdSettings
+    ) {
+        final String indexNames = concreteIndices.stream().map(Index::getName).collect(Collectors.joining(", "));
+        validateSearchNodes(currentState, indexNames);
+        validateDiskThresholdWaterMarkNotBreached(currentState, clusterInfo, diskThresholdSettings, indexNames);
+
+        final TieringValidationResult tieringValidationResult = new TieringValidationResult(concreteIndices);
+
+        for (Index index : concreteIndices) {
+            if (!validateHotIndex(currentState, index)) {
+                tieringValidationResult.addToRejected(index, "index is not in the HOT tier");
+                continue;
+            }
+            if (!validateRemoteStoreIndex(currentState, index)) {
+                tieringValidationResult.addToRejected(index, "index is not backed up by the remote store");
+                continue;
+            }
+            if (!validateOpenIndex(currentState, index)) {
+                tieringValidationResult.addToRejected(index, "index is closed");
+                continue;
+            }
+            if (!validateIndexHealth(currentState, index)) {
+                tieringValidationResult.addToRejected(index, "index is red");
+                continue;
+            }
+            tieringValidationResult.addToAccepted(index);
+        }
+
+        final Tuple<Set<Index>, Set<Index>> eligibleNodesCapacity = validateEligibleNodesCapacity(
```

Comment:
for `validateEligibleNodesCapacity` as well, we can pass in `tieringValidationResult` instance which can just update the rejected indices

Reply:
updated